plugins/modules/redfish_firmware_rollback.py

```diff
@@ -0,0 +1,360 @@
+    job_ids = []
+    for uri in preview_uri:
+        resp = redfish_obj.invoke_request("POST", update_uri, data={"ImageURI": uri})
+        time.sleep(30)
+        task_uri = resp.headers.get("Location")
+        task_id = task_uri.split("/")[-1]
+        job_ids.append(task_id)
+    return job_ids
+
+
+def rollback_firmware(redfish_obj, module, preview_uri, reboot_uri, update_uri):
+    current_job_status, failed_cnt, resetting = [], 0, False
+    job_ids = simple_update(redfish_obj, preview_uri, update_uri)
+    if module.params["reboot"] and preview_uri:
+        payload = {"ResetType": "ForceRestart"}
+        job_resp_status, reset_status, reset_fail = wait_for_redfish_reboot_job(redfish_obj, SYSTEM_RESOURCE_ID,
+                                                                                payload=payload)
+        if reset_status and job_resp_status:
+            job_uri = MANAGER_JOB_ID_URI.format(job_resp_status["Id"])
+            job_resp, job_msg = wait_for_redfish_job_complete(redfish_obj, job_uri)
+            job_status = job_resp.json_data
+            if job_status["JobState"] != "RebootCompleted" and job_msg:
+                module.fail_json(msg=JOB_WAIT_MSG.format(module.params["reboot_timeout"]))
+            elif job_status["JobState"] != "RebootCompleted" and not job_msg:
+                module.fail_json(msg=REBOOT_FAIL)
+        elif not reset_status and reset_fail:
+            module.fail_json(msg=reset_fail)
+
+        current_job_status, failed = get_job_status(redfish_obj, module, job_ids, job_wait=True)
+        failed_cnt += failed
+    if not module.params["reboot"] and preview_uri:
+        current_job_status, failed = get_job_status(redfish_obj, module, job_ids, job_wait=False)
+        failed_cnt += failed
+    if reboot_uri:
+        job_ids = simple_update(redfish_obj, reboot_uri, update_uri)
+        track, resetting, js_job_msg = wait_for_redfish_idrac_reset(module, redfish_obj, 900)
+        if not track and resetting:
+            reboot_job_status, failed = get_job_status(redfish_obj, module, job_ids, job_wait=True)
+            current_job_status.extend(reboot_job_status)
+            failed_cnt += failed
+    return current_job_status, failed_cnt, resetting
+
+
+def main():
+    specs = {
+        "name": {"required": True, "type": "str"},
+        "reboot": {"type": "bool", "default": True},
+        "reboot_timeout": {"type": "int", "default": 900},
+    }
+    specs.update(redfish_auth_params)
+    module = AnsibleModule(argument_spec=specs, supports_check_mode=True)
+    if module.params["reboot_timeout"] <= 0:
+        module.fail_json(msg=NEGATIVE_TIMEOUT_MESSAGE)
+    try:
+        with Redfish(module.params, req_session=True) as redfish_obj:
+            preview_uri, reboot_uri, update_uri = get_rollback_preview_target(redfish_obj, module)
+            job_status, failed, resetting = rollback_firmware(redfish_obj, module, preview_uri, reboot_uri, update_uri)
```

Comment:
update this variable name from failed to failed_count, as it seems we are comparing with string and int in line 337

Reply:
also have variable for len(job_status), as we are doing same thing in multiple time, better to call at once and store.

Reply:
Will update the ` failed_count`

Reply:
Updated